gdg/__init__.py

```diff
@@ -238,7 +276,35 @@ def list(self, gallery=""):
             images = images.order_by(SQL('path collate nocase'))
             result["images"] = [get_relative_path(baseurl, i.path) for i in images]
         return result
-    
+
+    @cherrypy.expose
+    @cherrypy.tools.allow(methods=['PUT', 'POST'])
+    def tag(self, **kwargs):
+        if not verify_key(kwargs.get('key', '')):
+            return "What's the magic word?"
+        tag_name = kwargs.get('tag', '')
+        if not tag_name:
+            return "You need to specify a tag name."
+        file_path = kwargs.get('file', '')
+        if not file:
+            return "You need to specify a file to tag"
+        image_folder = cherrypy.request.app.config['images']['path']
+        full_path = os.path.join(current_dir, image_folder, file_path)
+
+        dbpath = cherrypy.request.app.config['database']['path']
+        with GoddamnDatabase(dbpath) as db:
+            with db.transaction():
+                images = Image.select().where(Image.path == full_path)
+                if len(images) == 0:
```

Comment:
`images` at this point is still an object of the type `SelectQuery`, which is an iterator but not a list, so `len()` doesn't work.  Similarly you can't use an indexer on line 300 below.  For these to work you'd have to evaluate the query into a list, something like `images = [i for i in Image.select().where(Image.path == full_path)]`

Alternatively, since the path is practically the key for Images you can use `.count()` on the query object to test if any match the path and then use `get()` to get a single result if the count > 0.